test/TesterInternal/StreamingTests/SingleStreamTestRunner.cs

```diff
@@ -372,6 +374,54 @@ private void Heading(string testName)
             await StopProxies();
         }
 
+        public async Task StreamTest_22_TestImmutabilityDuringStreaming()
+        {
+            Heading("StreamTest_22_TestImmutabilityDuringStreaming");
+
+            IStreamingImmutabilityTestGrain itemProducer = GrainClient.GrainFactory.GetGrain<IStreamingImmutabilityTestGrain>(Guid.NewGuid());
+            string producerSilo = await itemProducer.GetSiloIdentifier();
+
+            // Setup consumers one in silo of item producer and the other one in another silo.
```

Comment:
Silo locality plays a role in the current implementation where all objects got wrapped as immutable.

If an item is transferred from a producer P to a local consumer CL and a remote consumer CR, then P and CL are accessing the same item while CR is only able to access a copy. So you have immutable items, that are not globally immutable.

The added behavior by setting `ENFORCE_ITEMS_IMMUTABILITY = false` allows to always deal with copies, even if producer and consumer are within the same silo. The fact that the item has been copied is validated via the emptyness of the `StreamImmutabilityTestObject.MyString` which is marked as `[NonSerialized]`.